### PR TITLE
Refactor and comment CodeContext and CodeParser.

### PR DESCRIPTION
**Pull request by troessner**

### CODE REVIEW

**Comment by troessner** [comment]
Starting with #395  - given the sensitive nature of the code we intend to refactor I'd rather go in baby steps here.:)

File: lib/reek/core/code_parser.rb [hunk 1]
```diff
@@ -9,24 +9,26 @@ module Core
     # Traverses a Sexp abstract syntax tree and fires events whenever
     # it encounters specific node types.
     #
-    # SMELL: This class is responsible for counting statements and for feeding
-    # each context to the smell repository.
-    # SMELL: This class has a name that doesn't match its responsibility.
     class CodeParser
-      def initialize(smell_repository, ctx = StopContext.new)
+      def initialize(smell_repository)
```
Comment (troessner) [inline lib/reek/core/code_parser.rb]: The second argument was never used anywhere.

File: lib/reek/examiner.rb [hunk 2]
```diff
@@ -16,9 +16,6 @@ class Examiner
     #
     # Creates an Examiner which scans the given +source+ for code smells.
     #
-    # The smells reported against any source file can be "masked" by
-    # creating *.reek files. See TBS for details.
-    #
```
Comment (troessner) [inline lib/reek/examiner.rb]: Misleading and unnecessary comment.

File: lib/reek/core/code_parser.rb [hunk 3]
```diff
@@ -9,24 +9,26 @@ module Core
     # Traverses a Sexp abstract syntax tree and fires events whenever
     # it encounters specific node types.
     #
-    # SMELL: This class is responsible for counting statements and for feeding
-    # each context to the smell repository.
-    # SMELL: This class has a name that doesn't match its responsibility.
```
Comment (mvz) [inline lib/reek/core/code_parser.rb]: These smells haven't been solved yet :smile:

**Comment by mvz** [comment]
Looks good. I'd just keep the SMELLS comments since they still need to be done.

**Comment by troessner** [comment]
I addressed your comment about the .....comment.:)
Merge?

**Pull request merged**